--- zapret2-0.9.4.4.orig/ip2net/Makefile
+++ zapret2-0.9.4.4/ip2net/Makefile
@@ -1,11 +1,8 @@
 CC ?= cc
 OPTIMIZE ?= -Os
-CFLAGS_PIC = -fPIC
-CFLAGS += -std=gnu99 $(OPTIMIZE) -flto=auto $(CFLAGS_PIC)
+CFLAGS += -std=gnu99 $(OPTIMIZE)
 CFLAGS_BSD = -Wno-address-of-packed-member
 CFLAGS_WIN = -static
-LDFLAGS_PIE = -pie
-LDFLAGS +=  $(LDFLAGS_PIE)
 
 LIBS = 
 LIBS_WIN = -lws2_32
--- zapret2-0.9.4.4.orig/nfq2/Makefile
+++ zapret2-0.9.4.4/nfq2/Makefile
@@ -1,18 +1,14 @@
 CC ?= cc
 PKG_CONFIG ?= pkg-config
 OPTIMIZE ?= -Os
-MINSIZE ?= -flto=auto -ffunction-sections -fdata-sections
+MINSIZE ?= -ffunction-sections -fdata-sections
 CFLAGS += -std=gnu99 $(OPTIMIZE) $(MINSIZE)
-CFLAGS_PIC = -fPIC
-CFLAGS_LINUX = -Wno-alloc-size-larger-than $(CFLAGS_PIC)
-CFLAGS_SYSTEMD = -DUSE_SYSTEMD
 CFLAGS_BSD = -Wno-address-of-packed-member $(CFLAGS_PIC)
 CFLAGS_CYGWIN = -Wno-address-of-packed-member -static -Wl,--nxcompat
 CFLAGS_CYGWIN32 =
 CFLAGS_CYGWIN64 = -Wl,--dynamicbase -Wl,--high-entropy-va
 CFLAGS_UBSAN = -fsanitize=undefined,alignment -fno-sanitize-recover=undefined,alignment
-LDFLAGS_PIE = -pie
-LDFLAGS += -flto=auto -Wl,--gc-sections $(LDFLAGS_PIE)
+LDFLAGS += -Wl,--gc-sections
 LDFLAGS_ANDROID = -llog
 STRIPP=-s
 LIBS =
--- zapret2-0.9.4.4.orig/nfq2/sec.c
+++ zapret2-0.9.4.4/nfq2/sec.c
@@ -14,7 +14,7 @@
 
 #include <sys/prctl.h>
 #include <sys/syscall.h>
-#include <linux/seccomp.h>
+//#include <linux/seccomp.h>
 #include <linux/filter.h>
 // __X32_SYSCALL_BIT defined in linux/unistd.h
 #include <linux/unistd.h>
@@ -113,6 +113,7 @@
 	filter->k = k;
 }
 // deny all blocked syscalls
+/*
 static bool set_seccomp(void)
 {
 #ifdef __X32_SYSCALL_BIT
@@ -142,6 +143,7 @@
 	set_filter(&prog.filter[idx++], BPF_JMP + BPF_JGT + BPF_K, 2 + BLOCKED_SYSCALL_COUNT, 0, 2); // 1 - stdout, 2 - stderr. greater are bad
 	set_filter(&prog.filter[idx++], BPF_LD + BPF_W + BPF_ABS, 0, 0, syscall_nr); // reload syscall_nr
 */
+/*
 	for(i=0 ; i<BLOCKED_SYSCALL_COUNT ; i++)
 	{
 		set_filter(&prog.filter[idx++], BPF_JMP + BPF_JEQ + BPF_K, BLOCKED_SYSCALL_COUNT-i, 0, blocked_syscalls[i]);
@@ -150,9 +152,12 @@
 	set_filter(&prog.filter[idx++], BPF_RET + BPF_K, 0, 0, SECCOMP_RET_KILL); // fail case
 	return prctl(PR_SET_SECCOMP, SECCOMP_MODE_FILTER, &prog) >= 0;
 }
+*/
 
 bool sec_harden(void)
 {
+	return false;
+/*
 	bool bRes = true;
 	if (prctl(PR_SET_NO_NEW_PRIVS, 1, 0, 0, 0))
 	{
@@ -168,11 +173,9 @@
 	}
 #endif
 	return bRes;
+*/
 }
 
-
-
-
 bool checkpcap(uint64_t caps)
 {
 	if (!caps) return true; // no special caps reqd
--- zapret2-0.9.4.4.orig/nfq2/sec.h
+++ zapret2-0.9.4.4/nfq2/sec.h
@@ -2,11 +2,12 @@
 
 #include <sys/types.h>
 #include <stdbool.h>
+#include <stdint.h>
 
 #ifdef __linux__
 
 #include <stddef.h>
-#include <sys/capability.h>
+#include <linux/capability.h>
 #include <linux/audit.h>
 
 bool checkpcap(uint64_t caps);
@@ -14,9 +15,9 @@
 int getmaxcap(void);
 bool dropcaps(void);
 
-#define syscall_nr (offsetof(struct seccomp_data, nr))
-#define arch_nr (offsetof(struct seccomp_data, arch))
-#define syscall_arg(x) (offsetof(struct seccomp_data, args[x]))
+//#define syscall_nr (offsetof(struct seccomp_data, nr))
+//#define arch_nr (offsetof(struct seccomp_data, arch))
+//#define syscall_arg(x) (offsetof(struct seccomp_data, args[x]))
 
 #ifndef __AUDIT_ARCH_64BIT
 #define __AUDIT_ARCH_64BIT 0x80000000
